File: eclipse35/source/strategyagent/src/main/sample_data/samples/scripts/process_data.rb
#
# author:anshul@marketcetera.com
# since 1.0.0
# version: $Id$
#
#
'require java'
include_class "org.marketcetera.strategy.ruby.Strategy"
include_class "org.marketcetera.marketdata.MarketDataRequest"

##################################################
# Strategy that processes market data via CEP    #
##################################################
class ProcessData < Strategy
    SYMBOLS = "AMZN,JAVA" # Depends on MD - can be other symbols
    CONTENT = "LATEST_TICK"
    MARKET_DATA_PROVIDER = "marketcetera" # Can also be activ, bogus, opentick
    CEP_QUERY = ["select t.symbolAsString as symbol, t.price * t.size as position from trade t"]
    CEP_PROVIDER = "esper"
    

    ##########################################
    # Executed when the strategy is started. #
    #                                        #
    # Use this method to set up data flows   #
    #  and other initialization tasks.       #
    ##########################################
    def on_start
      request = MarketDataRequest.newRequest().withSymbols(SYMBOLS).fromProvider(MARKET_DATA_PROVIDER).withContent(CONTENT)      
      request_processed_market_data(request, CEP_QUERY.to_java(:string), CEP_PROVIDER)
    end


    ############################################################
    # Executed when the strategy receives data of a type other #
    #  than the other callbacks                                #
    ############################################################
    def on_other(data)
      warn "Trade " + data.to_s
    end
end

File: root/tags/2.0.0/source/strategyagent/src/main/sample_data/samples/java/scripts/ProcessData.java
import org.marketcetera.strategy.java.Strategy;
import org.marketcetera.marketdata.MarketDataRequest;
import static org.marketcetera.marketdata.MarketDataRequest.Content;

/* $License$ */
/**
 * Strategy that processes market data via CEP.
 *
 * @author anshul@marketcetera.com
 * @version $Id$
 * @since 2.0.0
 */
public class ProcessData extends Strategy {
    private static final String [] SYMBOLS = {"AMZN", "JAVA"}; // Depends on MD - can be other symbols
    private static final String MARKET_DATA_PROVIDER = "marketcetera"; // Can be activ, bogus, marketcetera
    private static final String [] CEP_QUERY =
            {"select t.instrumentAsString as symbol, t.price * t.size as position from trade t"};
    private static final String CEP_PROVIDER = "esper";



    /**
     * Executed when the strategy is started.
     * Use this method to set up data flows
     * and other initialization tasks.
     */
    @Override
    public void onStart() {
        requestProcessedMarketData(MarketDataRequest.newRequest().
                withSymbols(SYMBOLS).
                fromProvider(MARKET_DATA_PROVIDER).
                withContent(Content.LATEST_TICK),
                CEP_QUERY, CEP_PROVIDER);
    }

    /**
     * Executed when the strategy receives any other event.
     *
     * @param inEvent the received event.
     */
    @Override
    public void onOther(Object inEvent) {
        warn("Trade " + inEvent);
    }
}
